pkg/resource/value_test.go

```diff
@@ -78,4 +80,16 @@ func TestValue_Pull(t *testing.T) {
 			t.Fatalf("Value (-want,+got)\n%s", diff)
 		}
 	})
+
+	t.Run("doesnt panic with no initial value", func(t *testing.T) {
+		val := NewValue()
+
+		res, err := val.Set(&traits.OnOff{State: traits.OnOff_OFF})
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		assert.True(t, proto.Equal(&traits.OnOff{State: traits.OnOff_OFF}, res))
```

Comment:
This test would produce better failure logs if you used the `cmp` package:
```go
if diff := cmp.Diff(&traits.OnOff{State: traits.OnOff_OFF}, res, protocmp.Transform()); diff != "" {
  t.Fatalf("Set response (-want,+got)\n%s", diff)
}
```
Like we use for other proto comparing tests.

Also means we can get rid of the testify dependency again